src/main/java/ca/etsmtl/leakageanalysisplugin/windows/ToolWindowFactory.java

```diff
@@ -18,95 +22,98 @@
-public class ToolWindowFactory implements com.intellij.openapi.wm.ToolWindowFactory, DumbAware
-{
+public class ToolWindowFactory implements com.intellij.openapi.wm.ToolWindowFactory, DumbAware {
     @Override
-    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow)
-    {
-        LeakageToolWindowContent toolWindowContent = new LeakageToolWindowContent(toolWindow);
+    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
+        MessageBusConnection busConnection = ProjectManager.getInstance().getDefaultProject().getMessageBus().connect();
+        LeakageToolWindowContent toolWindowContent = new LeakageToolWindowContent(toolWindow, busConnection);
         Content content = ContentFactory.getInstance().createContent(toolWindowContent.getContentPanel(), "", false);
         toolWindow.getContentManager().addContent(content);
     }
 
-    private static class LeakageToolWindowContent
-    {
+    private static class LeakageToolWindowContent {
         private final JPanel contentPanel = new JPanel();
-        private final ArrayList<LeakageTypeGUI> leakageTypes = new ArrayList<>();
+        private final ArrayList<LeakageTypeGUI> leakageTypeCounters = new ArrayList<>();
 
-        private LeakageApiService leakageApiService = new LeakageApiService();
-
-        public LeakageToolWindowContent(ToolWindow toolWindow)
-        {
-            leakageTypes.add(new LeakageTypeGUI("Overlap Leakage", "overlap leakage"));
-            leakageTypes.add(new LeakageTypeGUI("Multi-Test Leakage", "no independence test data"));
-            leakageTypes.add(new LeakageTypeGUI("Preprocessing Leakage", "pre-processing leakage"));
+        public LeakageToolWindowContent(ToolWindow toolWindow, MessageBusConnection busConnection) {
+            busConnection.subscribe(AnalyzeTaskListener.TOPIC, new AnalyzeTaskListener() {
+                @Override
+                public void updateResults(List<AnalysisResult> results) {
+                    LeakageToolWindowContent.this.updateResults(results);
+                }
+            });
+            leakageTypeCounters.add(new LeakageTypeGUI("Overlap Leakage", LeakageType.OVERLAP));
+            leakageTypeCounters.add(new LeakageTypeGUI("Multi-Test Leakage", LeakageType.MULTITEST));
+            leakageTypeCounters.add(new LeakageTypeGUI("Preprocessing Leakage", LeakageType.PREPROCESSING));
 
             contentPanel.setLayout(new BorderLayout(0, 20));
             contentPanel.setBorder(BorderFactory.createEmptyBorder(40, 0, 0, 0));
             contentPanel.add(createLeakagePanel(), BorderLayout.PAGE_START);
             contentPanel.add(createControlsPanel(toolWindow), BorderLayout.PAGE_END);
         }
 
+        private void updateResults(List<AnalysisResult> results) {
+            // TODO: FILL WINDOW TOOL WITH RESULTS (WILL NEED APPROPRIATE LAYOUT)
+            if (results.size() == 0) {
```

Comment:
Ça me semble plus clair comme ça.

```suggestion
            if (results.isEmpty()) {
```